### ISSUE TITLE
ERA001 false negative

### ISSUE THREAD

**Issue by JonathanPlasse** [astral-sh/ruff#3525]
```python
# if severity != logSeverity.DEBUG:
#    add_log(context, str(x), severity)

# with keyboard.Events() as events:
#     # Block for as much as possible
#     event = events.get(1e6)
#     if event.key == keyboard.KeyCode.from_char('q'):
#         print("goodbye")
#         raise KeyboardInterrupt
#     if event.key == keyboard.KeyCode.from_char('r'):
#         print("reloading servers...")
#         return
```
become
```python
# if severity != logSeverity.DEBUG:

# with keyboard.Events() as events:
#     # Block for as much as possible
#     if event.key == keyboard.KeyCode.from_char('q'):
#         raise KeyboardInterrupt
#     if event.key == keyboard.KeyCode.from_char('r'):
```
The upstream eradicate does not fix it either.

**Comment by JonathanPlasse** [comment]
```python
# with keyboard.Events() as events:
#     # Block for as much as possible
#     event = events.get(1e6)
#     if event.key == keyboard.KeyCode.from_char('q'):
#         print("goodbye")
#         raise KeyboardInterrupt
#     if event.key == keyboard.KeyCode.from_char('r'):
#         print("reloading servers...")
#         return
```
become
```python
# with keyboard.Events() as events:
#     # Block for as much as possible
#     if event.key == keyboard.KeyCode.from_char('q'):
#         raise KeyboardInterrupt
#     if event.key == keyboard.KeyCode.from_char('r'):
```
The upstream eradicate does not fix it either.

**Comment by charliermarsh** [comment]
Lots of spooky regex in there...

**Comment by calumy** [comment]
> I'm gonna merge with #3525.

This might be a typo as it is merging with itself.
